### PR TITLE
perf: optimize transform data structure

### PR DESCRIPTION
**Pull request by wumaolinmaoan**
## Summary by CodeRabbit

- **New Features**
	- Enhanced asset migration tools to better handle older data formats, ensuring compatibility.
	- Improved the handling of visual effects items, including adjustments to scale, size, and anchor points for more intuitive behavior.

- **Refactor**
	- Updated the `Transform` class to better manage visual properties such as size and anchor positioning, streamlining the integration of visual effects.

- **Bug Fixes**
	- Fixed issues related to the incorrect scaling and anchoring of sprites, ensuring they now display as intended.

### CODE REVIEW

File: packages/effects-core/src/transform.ts [hunk 1]
```diff
@@ -280,11 +281,10 @@ export class Transform implements Disposable {
    * @param y
    * @param z
```
Comment (yiiqii) [inline packages/effects-core/src/transform.ts:282]: 注释移除 param z

File: packages/effects-core/src/asset-migrations.ts [hunk 2]
```diff
@@ -272,3 +307,16 @@ export function version3Migration (scene: Record<string, any>): Scene {
 
   return scene as Scene;
 }
+
+/**
+ * 提取并转换 JSON 数据中的 anchor 值
+ */
+export function convertAnchor (anchor?: spec.vec2, particleOrigin?: spec.ParticleOrigin): spec.vec2 {
```
Comment (yiiqii) [inline packages/effects-core/src/asset-migrations.ts:314]: math 下的 convertAnchor 是不是要移除

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/asset-migrations.ts (hunk 2 above)
Comment (yiiqii) [reply packages/effects-core/src/asset-migrations.ts:314]: 我看还有其他地方有用这个方法，放在这里是否合适

**Review by RGCHN — DISMISSED** [review]

File: packages/effects-core/src/asset-migrations.ts (hunk 2 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/asset-migrations.ts:335]: 已移除

File: packages/effects-core/src/transform.ts (hunk 1 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/transform.ts]: 已修改

File: packages/effects-core/src/asset-migrations.ts (hunk 2 above)
Comment (yiiqii) [reply packages/effects-core/src/asset-migrations.ts:335]: > 已移除

math/util.ts 下有重复的 convertAnchor

**Review by yiiqii — APPROVED** [review]

**Pull request merged**